docs/concepts/tx-lifecycle.md

```diff
@@ -0,0 +1,176 @@
+
+The first step of consensus is the **block proposal**. One proposer amongst the validators is chosen by the consensus algorithm to create and propose a block - in order for `Tx` to be included, it must be in this proposer's mempool. 
+
+## State Changes
+
+The next step of consensus is to execute the transactions to fully validate them. All full-nodes that receive a block proposal execute the transactions by calling the ABCI functions [`BeginBlock`](./app-anatomy.md#beginblocker-and-endblocker), `DeliverTx` for each transaction, and [`EndBlock`](./app-anatomy.md#beginblocker-and-endblocker). While full-nodes each run everything individually, since the messages' state transitions are deterministic and transactions are explicitly ordered in the block proposal, this process yields a single, unambiguous result.
+
+```
+		-----------------------		
+		|Receive Block Proposal|        
+		-----------------------		
+		          |		
+			  v
+		-----------------------		
+		| BeginBlock	      |        
+		-----------------------		
+		          |		
+			  v			
+		-----------------------		    
+		| DeliverTx(tx0)      |  
+		| DeliverTx(tx1)      |   	  
+		| DeliverTx(tx2)      |  
+		| DeliverTx(tx3)      |  
+		|	.	      |  
+		|	.	      |
+		|	.	      |
+		-----------------------		
+		          |			
+			  v			
+		-----------------------
+		| EndBlock	      |         
+		-----------------------
+		          |			
+			  v			
+		-----------------------
+		| PreVote	      |         
+		-----------------------
+		          |			
+			  v			
+		-----------------------
+		| PreCommit	      |         
+		-----------------------
+		          |			
+			  v			
+		-----------------------
+		| Commit	      |         
+		-----------------------
+```
+
+### DeliverTx
+
+The `DeliverTx` ABCI function defined in [`baseapp`](./baseapp.md) does the bulk of the state change work: it is run for each transaction in the block in sequential order as committed to during consensus. Under the hood, `DeliverTx` is almost identical to `CheckTx` but calls the [`runTx`](./baseapp.md#runtx-and-runmsgs) function in deliver mode instead of check mode. Instead of using their `checkState` or `queryState`, full-nodes select a new copy, `deliverState`, to deliver `Tx`:
+
+* **Decoding:** Since `DeliverTx` is an ABCI call, `Tx` is received in the encoded `[]byte` form. Nodes first unmarshal the transaction, then call `runTx` in `runTxModeDeliver`, which is very similar to `CheckTx` but also executes and writes state changes.
+* **Checks:** Full-nodes call `validateBasicMsgs` and the `AnteHandler` again. This second check happens because they may not have seen the same transactions during the Addition to Mempool stage and a malicious proposer may have included invalid ones. One difference here is that the `AnteHandler` will not compare `gas-prices` to the node's `min-gas-prices`since that value is local to each node.
```

Comment:
Good, but add little explanation of why local values cannot be taken into account during `DeliverTx` (because state-changes must be deterministic, i.e. everyone must have the same state at the end of a block)